Controllers/comments-controller.js

```diff
@@ -0,0 +1,49 @@
+const { fetchArticleByID} = require("../Models/articles-model");
+const { fetchCommentsByID, insertComment } = require("../Models/comments-model");
+const { fetchUser } = require("../Models/users-model");
+
+function getCommentsByID(req, res, next) {
+    const { article_id } = req.params;
+
+    fetchArticleByID(article_id)
+    .then(()=>{
+       return fetchCommentsByID(article_id)
+    })
+    .then((comments)=>{
+        res.status(200).send(comments)
+    })
+    .catch((err)=>{
+        next(err);
+    });
+}
+
+function postComment(req, res, next){
+    const newComment = req.body;
+    const username = newComment.username;
+    const {article_id} = req.params;
+
+    function bodyCheck(newComment){
+        
+        if (!newComment.hasOwnProperty("username") ||!newComment.hasOwnProperty("body") || Object.keys(newComment).length !== 2) {
```

Comment:
This `if` statement and the `Promise.all` below are not necessary for this controller. In the seed function, the comments table has _author_ & _article_Id_ as _foreign keys_ that reference the `users` and `articles` tables respectively. 
As such, all you need for the query on this endpoint is the `INSERT INTO...`, if anything is missing or cannot be found in the database, PSQL will throw an error to the catch with a code of `23503` or `23502`